src/app/address/isEqualAddress.ts

```diff
@@ -12,13 +12,26 @@ export default function isEqualAddress(address1?: ComparableAddress, address2?:
     return isEqual(
         normalizeAddress(address1),
         normalizeAddress(address2)
-    );
+    ) && isSameState(address1, address2);
+}
+
+function isSameState(address1: ComparableAddress, address2: ComparableAddress): boolean {
+    const addressValues1 = [address1.stateOrProvince, address1.stateOrProvinceCode];
+    const addressValues2 = [address2.stateOrProvince, address2.stateOrProvinceCode];
+
+    if (addressValues1.some(value => addressValues2.includes(value) && !!value)) {
```

Comment:
I think you should only compare state code against state code and state name against state name. This line compares state code with state name and return true if they are the same.

> ...and for consignments, the stateOrProvince also accepts state codes as values

I don't think that is true. We automatically set the state code (`stateOrProvinceCode`) using the state name (`stateOrProvince`) if only the latter is provided. But we don't accept it as a state code if it is passed in as state name.

Reply:
I had the impression that we allowed that but after testing I found we actually don't. I have refactored accordingly. Thanks!